jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/JaxRsResourceBase.java

```diff
@@ -533,4 +535,48 @@ protected void verifyNumberOfElements(int actual, int expected, String errorMess
         Preconditions.checkArgument(actual == expected, errorMessage);
     }
 
+    protected Response createPaymentResponse(final UriInfo uriInfo, final Payment payment, final TransactionType transactionType, @Nullable final String transactionExternalKey) {
+        final PaymentTransaction createdTransaction = findCreatedTransaction(payment, transactionType, transactionExternalKey);
+        Preconditions.checkNotNull(createdTransaction, "No transaction of type '%s' found", transactionType);
+
+        switch (createdTransaction.getTransactionStatus()) {
+            case PENDING:
+            case SUCCESS:
+                return uriBuilder.buildResponse(uriInfo, PaymentResource.class, "getPayment", payment.getId());
+            case PAYMENT_FAILURE:
+                // 402 - Payment Required
+                return Response.status(402).build();
```

Comment:
I am actually not convinced about that one because the client has the ability to pass an `externalKey` which can be used to retrieve the payment. In theory `Location` header are specific to `201`, so unless we have good reason to twist http spec, i'd be more inclined to not return it (but i don't have a strong feeling, as long as we are consistent across our apis, which means that if we decide to return it, we should do a pass across all our apis to make sure this matches everywhere).